tests/t_batch_verifier.nim

```diff
@@ -0,0 +1,126 @@
+# Nim-BLSCurve
+# Copyright (c) 2018-Present Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import
+  # Standard library
+  unittest,
+  # internal
+  ../blscurve/openmp,
+  # Public API
+  ../blscurve
+
+# Tests for batch verification
+# Compile with -d:openmp for parallel tests
+proc omp_status(): string =
+  when defined(openmp):
+    "[Using OpenMP with " & $omp_get_max_threads() & " threads]"
+  else:
+    "[Serial]"
+
+template wrappedTest(desc: string, body: untyped): untyped =
+  ## Wrap test in a proc to avoid having globals everywhere
+  ## ballooning the test BSS space usage
+  ## properly test destructors/GC/try-finally, ...
+  ## aliasing
+  ## and optimizations (that don't apply to globals)
+  test desc:
+    proc wTest() =
+      body
+
+    wTest()
+
+proc keyGen(seed: uint64): tuple[pubkey: PublicKey, seckey: SecretKey] =
+  var ikm: array[32, byte]
+  ikm[0 ..< 8] = cast[array[8, byte]](seed)
+  let ok = ikm.keyGen(result.pubkey, result.seckey)
+  doAssert ok
+
+proc hash[T: byte|char](message: openarray[T]): array[32, byte] {.noInit.}=
+  result.bls_sha256_digest(message)
+
+proc inclExample(batcher: var BatchedBLSVerifier, seed: int, message: string) =
+  let (pubkey, seckey) = keyGen(seed.uint64)
+  let hashed = hash(message)
+  let sig = seckey.sign(hashed)
+  doAssert batcher.incl(pubkey, hashed, sig)
```

Comment:
The single-pubkey version can't return anything but `true` though.

Reply:
Same remark as https://github.com/status-im/nim-blscurve/pull/101#discussion_r538089219

Since other init/update/finish APIs also use a boolean I just used the same scheme.
Also BoringSSLOpenSSL also uses an always true return value for sha256 context updates.
If it's checked, API changes that lead to new errors, for example `incl` now includes subgroup checks will not lead to code change (but as they will lead to usage site re-review anyway it's a weak argument i guess).

Reply:
Ah, alright.

Curious if this is a case of cargo-culting on their part too, but it's harmless enough in the meantime.